### PR TITLE
Integrate Algolia search

### PR DESCRIPTION
**Pull request by ariel-pettyjohn**
**Description**

- Adds dependencies for Algolia search and necessary React library.

- Integrates React component into Nextra configuration. 

- Adds responsive styles for mobile search.

**Tests**

I've performed manual testing of search results to the best of my knowledge, and cross-browser testing of the React dependency. 

**Additional context**

In response to the bounty on Bountycaster: https://www.bountycaster.xyz/bounty/0x953ba9472714d4c20183c30f47df9ae7266c4a20

### CODE REVIEW

**Comment by sbvegan** [comment]
Awesome! Thanks for opening and I think I saw your email come through as well. To set expectations, our docs person is OOO this week. So earliest review would be sometime early next week.

**Comment by cpengilly** [comment]
@ariel-pettyjohn just reaching out to let you know I received the PR! I have some high priority docs to ship these next 2 weeks, but I'll circle back around to this PR the week of 6/17. Thanks again for submitting, and I'll be in touch soon! 💟

**Comment by ariel-pettyjohn** [comment]
@sbvegan, @cpengilly Thank you both for the updates! I'll keep an eye out around 6/17

**Comment by ariel-pettyjohn** [comment]
Heads up, I just realized I'm experiencing some technical difficulties with my work email. 

If you need to get ahold of me by email, you're welcome to use my personal address for the time being: ariel.pettyjohn@gmail.com :-)

**Comment by pydi99** [comment]
Hey optimism team @cpengilly , is the algolia search bounty still available?

**Comment by cpengilly** [comment]
Hi Ariel! Thanks for your patience and sorry for the delay! Reaching out to
let you know we'll be reviewing the submissions today and tomorrow. We'll
be in touch soon 💟

File: theme.config.tsx [hunk 1]
```diff
@@ -5,10 +5,25 @@ import { useConfig } from 'nextra-theme-docs'
...
 
+import { DocSearch } from '@docsearch/react';
+import '@docsearch/css';
+
 const config: DocsThemeConfig = {
+  search: { 
+    component: <DocSearch 
+      // search api key, okay to be public
+      // https://support.algolia.com/hc/en-us/articles/18966776061329-Can-the-search-API-key-be-public
+      apiKey="702dd042cad3dfd9581e362b36033291"
```
[the remark this answers is not in the document]
Comment (sbvegan) [reply theme.config.tsx:16]: I know rabbit, I had to go against my instincts for this one. Algolia's search api is okay to be public.

**Comment by cpengilly** [comment]
@ariel-pettyjohn Thanks for submitting! We have two small requests to finalize the project: 🎉 
* Everything is functional on the frontend but no event data is being sent to the Algolia back end dashboard. Please check out [this link ](https://www.algolia.com/doc-beta/libraries/search-insights/events/media) for docs on sending event data through using the Search Only API.
* The search bar has a clean design for the light mode version of the site, but could use some cleanup for darkmode. See the image below. You can change from dark/light mode using the switch on the bottom left section of the site.
dark-mode-search

Just a side note: You may need to pull down the latest version of the PR before making changes.
Thanks and look forward to closing this out soon! 💟

**Comment by cpengilly** [comment]
> Hey optimism team @cpengilly , is the algolia search bounty still available?

The bounty is open until Fri 6/28. Additional details are [here](https://oplabs.notion.site/Algolia-Search-Bounty-Spec-b36573a3d36643c4b40eb6568b476f44).

**Comment by ariel-pettyjohn** [comment]
Hey there @cpengilly.

I just pushed some changes for your review:

- I streamlined the search bar style for dark mode to mirror the style in light mode.

- I also initialized Search Insights, and added an example to the docs on how to add click tracking. I didn't add any specific click/conversion tracking because I'm not sure what your goals are, but I was able to verify that events are now being sent (which you can see in the screenshot).

Thanks and let me know what you think!

Events

**Comment by ariel-pettyjohn** [comment]
Hey team! 

One final note I forgot to mention: I did look into the option of managing the index manually and considered 2 choices.

1) was to scrape the website itself after build, and 2) was to extract text from the .mdx files at/before build. 

I've spent a lot of time scraping websites throughout my career, and, knowing how brittle the process is, I'm positive that Algolia's engineers will deliver better results than choice 1.

I also spent some time exploring choice 2 because it seemed to be more accurate and less brittle than choice 1 at first, and I was able to find solutions to extract text from markdown. The problem is that extracting text from .mdx in particular isn't a well-solved problem the way it is for plain markdown.

So, from my experience, I believe letting Algolia manage the index is going to give you the best search results, with the least cost and effort on your part, while also sparing you from having to invent an additional solution to extract text from .mdx.

Cheers and I hope you all have a great holiday!

**Comment by ariel-pettyjohn** [comment]
Hey there @cpengilly, any further notes or revisions? Seems like we're at the finish line here 👍

**Comment by ariel-pettyjohn** [comment]
Quick note:

I just noticed that Algolia moved the crawler interface to its own domain and buried the link deep in the settings of the primary dashboard. 

Just in case you haven't found it, you can access the crawler interface for the DocSearch index at https://crawler.algolia.com/ using the same credentials as the primary Algolia dashboard.

Cheers!

**Comment by cpengilly** [comment]
Thank you! We will be in touch by end of July with all bounty submitters on the results. 💟

**Comment by ariel-pettyjohn** [comment]
Hola, I see there are still a couple of unsuccessful checks. Those look like the linter being picky on your end, but please let me know if there's anything I need to do on my end to help resolve those 👍

**Comment by cpengilly** [comment]
Thank you for participating in our recent bounty challenge. We appreciate the time and effort you put into your submission.
We've completed our review and after careful consideration, we have selected another submission for the reward.

We value your contribution and encourage you to keep an eye out for future opportunities. We’d love to see your work in our upcoming challenges! 💟

**Pull request closed without merging**